### PR TITLE
added event for validation blur

### LINKED ISSUE #4650: Required field stays green for file option on PDP if user uploaded file and then canceled the second try to upload file
**Description**:
Required field stays green for file option on PDP if user uploaded file and then canceled the second try to upload file

**Expected behavior**:
File field becomes red because file isn't selected for required option
Screenshot 2022-04-04 at 13 03 56

**To Reproduce**:
1. Open product with required file option - https://master.d25kd40decjpue.amplifyapp.com/default/women/women-shoes/flip-flops/havaianas-green.html
2. Click on file input -> Select file -> Upload file
3. Click on file input -> Click on cancel

**Actual result**:
The file field stays green 
https://drive.google.com/file/d/17wExFci0c3AsXQQXNZ-GjJWigddcjDZJ/view?usp=sharing

### LINKED ISSUE #4715: Safari, FF: file is not deleted from filed if user clicked on cancel after previous successful loading
**Description**:
Safari, FF: file is not deleted from filed if user clicked on cancel after previous successful loading

**Expected behavior**:
Safari, FF: file is deleted from filed if user clicked on cancel

**Screenshots, Video, Logs**
https://vimeo.com/700694391/ed750f4ac0

**To Reproduce**:
1. Open https://master.d25kd40decjpue.amplifyapp.com/default/testing-simple.html
2. Click on field for adding file
3. Select some valid file
4. Click 'Upload'
5. Click on file field > Cancel

**Actual result**:
Safari, FF: file is not deleted from filed if user clicked on cancel after previous successful loading

### PR DESCRIPTION
**Pull request by AzizKHAN030**
**Related issue(s):**
* Fixes #4650 
* Fixes #4715 

**Problem:**
* Field stayed green if the user decided to remove the selected file
* In issue #4715 the FF didn't support the onChange event on cancel

**In this PR:**
* Added blur event for validation 
* For #4715 implemented new method cancelFileField, and cleaning the state and force validating to get error because it didn't validate automatically

### CODE REVIEW

**Review by zans-laksa — APPROVED** [review]

**Review by zans-laksa — APPROVED** [review]

File: packages/scandipwa/src/component/Field/Field.container.js [hunk 1]
```diff
@@ -139,7 +139,8 @@ export class FieldContainer extends PureComponent {
             ? !!this.fieldRef.checked
             : this.fieldRef.value;
         const newValidRule = this.handleShowLengthError();
-        const response = validate(type === FIELD_TYPE.file
+        // eslint-disable-next-line max-len
+        const response = data === 'errorRequired' ? validate('', { isRequired: true }) : validate(type === FIELD_TYPE.file
```
Comment (zans-laksa) [inline packages/scandipwa/src/component/Field/Field.container.js:143]: Please don't disable the eslint without a reason.
Split the line in this case.

Comment (zans-laksa) [reply packages/scandipwa/src/component/Field/Field.container.js:143]: Actually, this line looks too complex.
Please refactor it. Maybe move it to the separate method and use the conditional block there.

**Comment by zans-laksa** [comment]
We were speaking about this today and decided to create a field reset function for all fields.
The to use it to handle file dialog cancel even issue on FF.
I don't see this solution in this PR.

**Review by zans-laksa — CHANGES_REQUESTED** [review]

**Comment by AzizKHAN030** [comment]
I created method in parentConatainer `Field.container` and passed it to the `FileField.component`, I'm binding and passing the child context on click. Resetting the child state and ref value(for proper validation) from the parent container. If desired it can be used for other fields also by passing context and customizing for other situations

File: packages/scandipwa/src/component/Field/Field.component.js [hunk 2]
```diff
@@ -123,10 +125,18 @@ export class Field extends PureComponent {
     }
 
     renderFile() {
-        const { attr, events, setRef } = this.props;
+        const {
+            attr, events, setRef, validate, resetFieldValue
```
Comment (zans-laksa) [inline packages/scandipwa/src/component/Field/Field.component.js:129]: When there are more than two items in the list, please put them in one column.

File: packages/scandipwa/src/component/Field/Field.container.js [hunk 3]
```diff
@@ -120,6 +120,15 @@ export class FieldContainer extends PureComponent {
         this.setState({ validationResponse: null });
     }
 
+    resetFieldValue(fieldContext) {
+        fieldContext.setState({
+            value: '', fileName: '', isLoading: false
```
Comment (zans-laksa) [inline packages/scandipwa/src/component/Field/Field.container.js:125]: When there are more than two items in the list, please put them in one column.

File: packages/scandipwa/src/component/FieldFile/FieldFile.component.js [hunk 4]
```diff
@@ -92,6 +94,7 @@ export class FieldFile extends PureComponent {
                   { ...attr }
                   // eslint-disable-next-line @scandipwa/scandipwa-guidelines/jsx-no-props-destruction
                   { ...events }
+                  onClick={ resetFieldValue }
```
Comment (zans-laksa) [inline packages/scandipwa/src/component/FieldFile/FieldFile.component.js:97]: Do you override the `click` event here?
What if we pass `onClick` prop through the `events` config?

Comment (zans-laksa) [reply packages/scandipwa/src/component/FieldFile/FieldFile.component.js:97]: You pass `onClick` inside the component.
But maybe you could put it to the `events` config inside the container?
And please make a check - if onClick already is present, call it inside the new event handler.

Comment (zans-laksa) [reply packages/scandipwa/src/component/FieldFile/FieldFile.component.js:97]: Please see the `onChange` handling inside the container for the example.

**Review by zans-laksa — CHANGES_REQUESTED** [review]

File: packages/scandipwa/src/component/FieldFile/FieldFile.component.js (hunk 4 above)
Comment (AzizKHAN030) [reply packages/scandipwa/src/component/FieldFile/FieldFile.component.js]: Done!

**Review by zans-laksa — APPROVED** [review]

**Pull request merged**